src/ServerScriptService/GravityController/GravityController/Collider.lua

```diff
@@ -158,11 +162,15 @@ function ColliderClass:Update(force, cframe)
 end
 
 function ColliderClass:IsGrounded(isJumpCheck)
-	local parts = isJumpCheck and self.JumpDetector:GetTouchingParts() or self._floorTouchingParts
-	for _, part in pairs(parts) do
-		if not part:IsDescendantOf(self.Controller.Character) and part.CanCollide then
-			return true
+	if isJumpCheck then
+		local parts = self.JumpDetector:GetTouchingParts()
+		for _, part in pairs(parts) do
+			if not part:IsDescendantOf(self.Controller.Character) and part.CanCollide then
+				return true
+			end
 		end
+	else
+		return #self._floorTouchingParts > 0
```

Comment:
since the input signal for floorTouchingParts has already been filtered, we can now simply check if the floorTouchingParts is non empty